src/nvme/nbft.c

```diff
@@ -0,0 +1,739 @@
+// SPDX-License-Identifier: LGPL-2.1-or-later
+/*
+ * This file is part of libnvme.
+ * Copyright (c) 2021-2022, Dell Inc. or its subsidiaries.  All Rights Reserved.
+ *
+ * Authors: Stuart Hayes <Stuart_Hayes@Dell.com>
+ *
+ */
+
+#include <errno.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+#include <arpa/inet.h>
+#include <ccan/endian/endian.h>
+
+#include "private.h"
+#include "nbft.h"
+#include "log.h"
+
+
+#define MIN(a, b) (((a) < (b)) ? (a) : (b))
+
+static __u8 csum(void *buffer, int length)
+{
+	int n;
+	__u8 sum = 0;
+
+	for (n = 0; n < length; n++)
+		sum = (__u8)(sum + ((__u8 *)buffer)[n]);
+	return sum;
+}
+
+static void format_ip_addr(char *buf, size_t buflen, __u8 *addr)
+{
+	struct in6_addr *addr_ipv6;
+
+	addr_ipv6 = (struct in6_addr *)addr;
+	if (addr_ipv6->s6_addr32[0] == 0 &&
+	    addr_ipv6->s6_addr32[1] == 0 &&
+	    ntohl(addr_ipv6->s6_addr32[2]) == 0xffff)
+		/* ipv4 */
+		inet_ntop(AF_INET, &(addr_ipv6->s6_addr32[3]), buf, buflen);
```

Comment:
No need to check if the `inet_ntop` is successful?

Reply:
fixed with my latest update.

Reply:
> No need to check if the `inet_ntop` is successful?

The answer to this question is: NO we don't want to check if `inet_ntop` is successful.  Doing so will cause the `read_hfi_info_tcp` function to inappropriately fail because many of the different ip addressed in the nbft are blank.  I've reverted patch https://github.com/linux-nvme/libnvme/pull/572/commits/8411348419752d0dd511bd32bdee7fe398796129 from this pull request as testing with this change failed.  I am not able to boot with NVMe/TCP with this patch.

Reply:
Ah okay, I am looking at the diff right now and wondered what happened with this code.